metrics/sockets.go

```diff
@@ -0,0 +1,33 @@
+package metrics
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+// getSocketCount returns the number of open sockets
+func getSocketCount() uint64 {
+	var socketCount uint64
+	pid := os.Getpid()
+	base := fmt.Sprintf("/proc/%d/fd", pid)
+	fds, err := ioutil.ReadDir(base)
+	if err != nil {
+		// Should we log something here?
```

Comment:
what's the downside of logging something? might help us diagnose why weren't having problems